src/view.ts

```diff
@@ -0,0 +1,62 @@
+import { Graph, Vertex } from "./graph.ts";
+
+export function createView(graph: Graph) {
+  return Object.entries(graph.roots).reduce((view, [name, root]) => {
+    return {
+      ...view,
+      [name]: Object.values(root).reduce((entries, v) => {
+        return {
+          ...entries,
+          [v.id]: createVertexView(v, graph),
+        }
+      }, {} as Record<number, VertexView>)
+    };
+  }, {} as GraphView);
```

Comment:
This wasn't in the original implementation, but we should make this `Object.create()` so that we can actually make the list of entries dynamic so that new vertices will show up in between reads.